extension/src/parser.js

```diff
@@ -181,8 +181,27 @@ const getTargeting = ad => {
 // restore the state of the page when the extension clicks around.
 const refocus = cb => {
   const focus = document.activeElement;
+  const ranges = [];
+  for (var i = 0; i < window.getSelection().rangeCount; i++) {
+    ranges.push([
+      window.getSelection().getRangeAt(i).startContainer,
```

Comment:
Just save the output of `window.getSelection().getRangeAt(i)` and reference `startContainer` etc below, right?

Reply:
Under some circumstances, the range object gets mutated.

Reply:
K, then only do one call to getSelection, we’re perf stretched as is.

Reply:
good call